dev-util/visual-studio-code/visual-studio-code-1.22.2.ebuild

```diff
@@ -0,0 +1,98 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit eutils xdg-utils
+
+DESCRIPTION="Microsoft's free and open-source code editor."
+HOMEPAGE="https://code.visualstudio.com/"
+SRC_URI="https://github.com/Microsoft/vscode/archive/${PV}.tar.gz -> ${P}.tar.gz
+	https://www.hlmjr.com/vscode-assets.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND="net-libs/nodejs
+	sys-apps/yarn"
+RDEPEND="${DEPEND}
+	dev-libs/nss
+	dev-vcs/git
+	gnome-base/gconf
+	media-libs/alsa-lib
+	x11-libs/gtk+:2
+	x11-libs/libnotify
+	x11-libs/libxkbfile
+	x11-libs/libXScrnSaver
+	x11-libs/libXtst"
+
+S="${WORKDIR}"/vscode-"${PV}"
+
+pre_build_checks() {
+	CHECKREQS_MEMORY="2G" # due to max_old_space_size node setting below
+}
+
+src_prepare() {
+	eapply_user
+}
+
+src_compile() {
+	pushd "${S}"
+
+	# MS writes to different output directory based on arch w/different naming scheme
+	if use amd64; then
+		OUTPUT_ARCH=x64
+	elif use x86; then
+		OUTPUT_ARCH=ia32
+	fi
+
+	ulimit -n 8192 || die # avoid git errors >=1.22.0
+	yarn install --arch=${OUTPUT_ARCH}
+
+	# adjust max_old_space_size because default may be too low for node to build
+	/usr/bin/node --max_old_space_size=2048 node_modules/gulp/bin/gulp.js vscode-linux-${OUTPUT_ARCH}
+}
+
+src_install() {
+	# MS writes to different output directory based on arch w/different naming scheme
+	if use amd64; then
+		OUTPUT_ARCH=x64
+	elif use x86; then
+		OUTPUT_ARCH=ia32
+	fi
+
```

Comment:
`OUTPUT_ARCH` is a global var in your case and is saved between phases, so it should be available here, no need to re-define.

Reply:
Removed.